### DISCUSSION TITLE
Restauring volume backup as a Dynamic PVC

### DISCUSSION THREAD

**Discussion by vinid223** [longhorn/longhorn#3148] [Q&A] [answered]
Hello,

I had the unfortunate issue of having to restore some backups I have made. 

All went well and everything went back up. But now, the volumes are starting to get full and I want to expand them. 

Since the volumes were initially created using a PVC definition in yaml, I normally update the PVC yaml to the new volume size and redeploy with `kubectl apply`. But since I restaured the volumes and created them using Longhorn instead of my original volume definition from the backups, they are now not considered a dynamic PVC and I can't resize them anymore.

So here are the questions I have:

Is there a way to convert the volumes to be able to expand them ? 
If it is not possible, what would be the best approach to create a new PVC and move data without too much file manipulations between PVC ?
Should have I done the backup restoration process differently to prevent the issue I am having ?

**Comment by PhanLe1010** [comment, marked as answer]
I think the answer for all of your 3 questions is:

* Check the spec of the PVC to see what is the StorageClass name. By default, Longhorn UI set it to `longhorn-static`
* Create a new StorageClass with the above name. See [here](https://longhorn.io/docs/1.2.2/references/examples/#storageclass) for an example. Remember to set the correct parameter in the SC corresponding to the volume you are having. For example, if the volume is having ext4 fs, set the `parameters.fsType: ext4` in the SC

After this, the volume should be able to be expanded. 


We are working to enhance this user experience. The effort is tracked at https://github.com/longhorn/longhorn/issues/2723
